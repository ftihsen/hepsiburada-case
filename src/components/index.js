import Header from './Header/Header';
import SubHeader from './Header/SubHeader';
import Card from './Card/Card';
import CardImage from './Card/CardImage';
import CardProductTitle from './Card/CardProductTitle';
import Button from './Button/Button';
import Badge from './Badge/Badge';
import SearchInput from './Search/SearchInput';
import BasketModal from './Basket/BasketModal';
import BasketModalCard from './Basket/BasketModalCard';
import SideBar from './SideBar/SideBar';
import CardProductFilters from './Card/CardProductFilters';
import CardProductPrice from './Card/CardProductPrice';
import SelectBox from './SelectBox/SelectBox';
import Modal from './Modal/Modal';
import ProducList from './ProducList/ProducList';
import Logo from './Logo/Logo';

export {
  Header,
  SubHeader,
  Card,
  CardProductTitle,
  BasketModalCard,
  Button,
  Badge,
  SearchInput,
  BasketModal,
  CardImage,
  SideBar,
  CardProductFilters,
  CardProductPrice,
  SelectBox,
  Modal,
  ProducList,
  Logo,
};
