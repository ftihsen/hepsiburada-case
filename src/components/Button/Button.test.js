import { render, fireEvent } from '@testing-library/react';
import { Button } from 'components';

describe('Button Components', () => {
  it('should render with children prop', () => {
    const title = 'Submit';
    const { getByText } = render(<Button>{title}</Button>);

    expect(getByText(title)).toBeTruthy();
  });

  it('should increment number on click', () => {
    let number = 1;
    const { getByText } = render(
      <Button
        onClick={() => {
          number += 1;
        }}
      >
        Button
      </Button>,
    );

    fireEvent.click(getByText('Button'), new MouseEvent('click'));

    expect(number).toBe(2);
  });

  it('should match snapshot', () => {
    const title = 'Submit';
    const { container } = render(<Button>{title}</Button>);

    expect(container).toMatchSnapshot();
  });
});
