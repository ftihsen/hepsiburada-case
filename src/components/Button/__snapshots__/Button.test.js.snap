// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Components should match snapshot 1`] = `
<div>
  <button
    class="button button-undefined"
  >
    Submit
  </button>
</div>
`;
